### PR TITLE
IDEMPIERE-1604 - Extend features for Window Customization

### PR DESCRIPTION
**Pull request by ralexsander**
This will extend the window customization functionality.
Single Row layout and Read Only fields on window customization are now Lists, so it can be empty. This way the default behaviour will be preserved. You are now able to customize the WHERE clause (it will be appended to original WHERE clause for security reasons), process, order by, display logic and read only logic

### CODE REVIEW

File: migration/i7.1z/oracle/201809141624_IDEMPIERE-1604.sql [hunk 1]
```diff
@@ -0,0 +1,162 @@
+SET SQLBLANKLINES ON
+SET DEFINE OFF
+
+-- IDEMPIERE-1604 Improvement on Window Customization
+-- 14/09/2018 14h56min16s BRT
+INSERT INTO AD_Column (AD_Column_ID,Version,Name,Description,Help,AD_Table_ID,ColumnName,FieldLength,IsKey,IsParent,IsMandatory,IsTranslated,IsIdentifier,SeqNo,IsEncrypted,AD_Reference_ID,AD_Client_ID,AD_Org_ID,IsActive,Created,CreatedBy,Updated,UpdatedBy,AD_Element_ID,IsUpdateable,IsSelectionColumn,EntityType,IsSyncDatabase,IsAlwaysUpdateable,IsAutocomplete,IsAllowLogging,AD_Column_UU,IsAllowCopy,SeqNoSelection,IsToolbarButton,IsSecure) VALUES (213691,0,'Sql WHERE','Fully qualified SQL WHERE clause','The Where Clause indicates the SQL WHERE clause to use for record selection. The WHERE clause is added to the query. Fully qualified means "tablename.columnname".',466,'WhereClause',2000,'N','N','N','N','N',0,'N',14,0,0,'Y',TO_DATE('2018-09-14 14:56:16','YYYY-MM-DD HH24:MI:SS'),100,TO_DATE('2018-09-14 14:56:16','YYYY-MM-DD HH24:MI:SS'),100,630,'Y','N','D','N','N','N','Y','9cd3e72b-9a57-4744-9d79-386e9473e573','Y',0,'N','N')
+;
+
+-- 14/09/2018 14h56min26s BRT
+ALTER TABLE AD_UserDef_Tab ADD WhereClause VARCHAR2(2000) DEFAULT NULL
```
Comment (CarlosRuiz-globalqss) [inline migration/i7.1z/oracle/201809141624_IDEMPIERE-1604.sql:10]: Please change this to:
VARCHAR2(2000 CHAR)

And same for
AD_UserDef_Tab.OrderByClause
AD_UserDef_Tab.DisplayLogic
below

**Review by CarlosRuiz-globalqss — CHANGES_REQUESTED** [review]

File: migration/i7.1z/oracle/201809141624_IDEMPIERE-1604.sql (hunk 1 above)
Comment (ralexsander) [reply migration/i7.1z/oracle/201809141624_IDEMPIERE-1604.sql]: Done

**Comment by CarlosRuiz-globalqss** [comment]
Thanks @ralexsander - can you please also add this change to the oracle script?

[pr274.txt](https://github.com/idempiere/idempiere/files/5291982/pr274.txt)

**Comment by ralexsander** [comment]
> Thanks @ralexsander - can you please also add this change to the oracle script?
> 
> [pr274.txt](https://github.com/idempiere/idempiere/files/5291982/pr274.txt)

Done @CarlosRuiz-globalqss

**Review by CarlosRuiz-globalqss — APPROVED** [review]

**Pull request merged**